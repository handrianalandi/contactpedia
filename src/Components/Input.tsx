import styled from '@emotion/styled'
import React, { forwardRef } from 'react'
const InputComponent = styled.input`
    border: 1px solid #ccc;
    border-radius: 15px;
    padding: 10px;
    width: 100%;
    margin: 0 auto;
    display: block;
    box-sizing: border-box;

    &:focus{
        outline: none;
        border: 2px solid #06BA63;
    }
`
interface InputProps {
    onChange?: (e: React.ChangeEvent<HTMLInputElement>) => void;
    value?: string;
    placeholder?: string;
    isRequired?: boolean;
    type?: string;
}

const Input = forwardRef<HTMLInputElement, InputProps>(
    ({ onChange = ()=>{}, placeholder,isRequired=false,type='text', ...restProps }, ref) => (
        <InputComponent
            ref={ref}
            placeholder={isRequired?`${placeholder}*`:placeholder}
            onChange={onChange}
            data-testid="input-element-test"
            {...restProps}
        />
    )
);

export default Input;