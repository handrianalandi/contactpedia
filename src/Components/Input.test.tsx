import {render,fireEvent,screen} from '@testing-library/react';
import Input from './Input';
describe('Input', () => {
    const props = {
        mockOnChange: jest.fn(),
        placeholder: 'test',
        value: 'test',
        type: 'text',
    }
    it('should render correctly without props', () => {
        const {container} = render(<Input />);
        expect(container).toMatchSnapshot();
    }
    );

    it('should render correctly with props', () => {
        const {container} = render(<Input {...props}/>);
        expect(container).toMatchSnapshot();
    }
    );

    it('should call onChange when input is changed', () => {
        const mockOnChange = jest.fn();
        render(<Input onChange={mockOnChange}/>);
        //get the input
        const input = screen.getByTestId('input-element-test');
        //fire the change event
        fireEvent.change(input, {target: {value: 'test'}});
        expect(mockOnChange).toHaveBeenCalled();
    }
    );
});