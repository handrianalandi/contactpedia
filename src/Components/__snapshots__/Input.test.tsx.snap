// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should render correctly with props 1`] = `
.emotion-0 {
  border: 1px solid #ccc;
  border-radius: 15px;
  padding: 10px;
  width: 100%;
  margin: 0 auto;
  display: block;
  box-sizing: border-box;
}

.emotion-0:focus {
  outline: none;
  border: 2px solid #06ba63;
}

<div>
  <input
    class="emotion-0"
    data-testid="input-element-test"
    placeholder="test"
    type="text"
    value="test"
  />
</div>
`;

exports[`Input should render correctly without props 1`] = `
.emotion-0 {
  border: 1px solid #ccc;
  border-radius: 15px;
  padding: 10px;
  width: 100%;
  margin: 0 auto;
  display: block;
  box-sizing: border-box;
}

.emotion-0:focus {
  outline: none;
  border: 2px solid #06ba63;
}

<div>
  <input
    class="emotion-0"
    data-testid="input-element-test"
    type="text"
  />
</div>
`;
